test/src/e2e/test_datatypes.py

```diff
@@ -0,0 +1,498 @@
+import datetime
+import decimal
+import math
+import uuid
+
+import pytest
+
+from util import pyodbc_connection, create_table, rows_as_values
+
+
+# FIXME: None is converted to an empty string (probably by PyODBC itself?)
+#  After the fix, Nullable test cases should be re-added
+#  Sample error: Attempt to read after eof: while converting '' to UInt8. (ATTEMPT_TO_READ_AFTER_EOF)
+#
+# TODO:
+#  Bool
+#  (U)Int128
+#  (U)Int256
+#  Decimal256
+#  DateTime64
+#  Array
+#  Tuple
+#  Map
+#  LowCardinality
```

Comment:
Note that in the deleted [regression.py](https://github.com/ClickHouse/clickhouse-odbc/blob/2f55d016420af9dfbd503ba122c0c1549c180363/test/parameterized/regression.py), nullable tests are also marked as expected failure.
However, most of the data types tests that were "not supported" actually work, and implemented now.